src/lib/drivers/gyroscope/PX4Gyroscope.cpp

```diff
@@ -122,11 +125,19 @@ void PX4Gyroscope::set_device_type(uint8_t devtype)
 
 void PX4Gyroscope::set_update_rate(uint16_t rate)
 {
-	_update_rate = rate;
-	const uint32_t update_interval = 1000000 / rate;
+	_update_rate = math::constrain((int)rate, 50, 32000);
+	const uint32_t update_interval = 1000000 / _update_rate;
+
+	// constrain IMU integration time 1-20 milliseconds (50-1000 Hz)
+	int32_t imu_integration_time_us = math::constrain(_param_imu_integ_time.get(), 1000, 20000);
+
+	if (imu_integration_time_us != _param_imu_integ_time.get()) {
+		_param_imu_integ_time.set(imu_integration_time_us);
+		_param_imu_integ_time.commit_no_notification();
```

Comment:
Isn't it a problem that all Gyroscope and Accelerometer drivers would try to correct the out of range parameter more or less at the same time?

Reply:
I would at least define the [1000, 20000] interval shared to make sure it's always in sync.

Reply:
Yes I didn't like that part either. Very soon (early in the v1.12 release cycle) the plan is to gut all of this code from PX4Accelerometer/PX4Gyroscope and move it "downstream" to sensors/vehicle_imu where we can effectively synchronize accel + gyro.